### PR TITLE
KT-9669: Join Lines should add semicolon when joining statements into the same line

### PR DESCRIPTION
**Pull request by voddan**
#KT-1234 Fixed: introduced `JoinStatementAddSemicolonHandler`; registered the handler in `plugin.xml`; added test coverage

### CODE REVIEW

File: idea/testData/joinLines/addSemicolon/WhileAndACall.kt.after [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fun foo() {
+    while (true) println();<caret> println()
```
Comment (piotrek1543) [inline idea/testData/joinLines/addSemicolon/WhileAndACall.kt.after]: why did you put there semicolon?

File: idea/testData/joinLines/addSemicolon/WhileAndACall.kt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+fun foo() {
+    <caret>while (true) println()
```
Comment (piotrek1543) [inline idea/testData/joinLines/addSemicolon/WhileAndACall.kt]: how do you want to interrupt this infinite loop statement?

Comment (voddan) [inline idea/testData/joinLines/addSemicolon/WhileAndACall.kt]: Not sure what you mean. Why should I want to interrupt it?

**Comment by yole** [comment]
Thanks a lot for your contribution!

**Comment by voddan** [comment]
Funny thing is I had to learn half of the Kotlin compiler before I found the simpler solution. Took me long enough 😄

**Comment by voddan** [comment]
BTW, the build process for the project is just horrible. 

I spent a day and a half in the beginning before I could compile tests without issues. Even worse, IDE stoped compiling after I rebased to the current master. Deleting the folder and downloading everything from scratch did not help. 

Please do something about it. Migrating everything to a proper Maven project could definitely help.

**Comment by yole** [comment]
We're in the process of migrating the build of Kotlin to gradle-kotlin-dsl.

**Comment by voddan** [comment]
Glad to hear that! 

Please also migrate the downloading of the dependancies, as well as the build targets 🙏  Updating dependancies was a nightmare over unstable internet, I had to hack the script and download archives by hand. Build targets, dependent on the IJ configuration, is a very unstable solution. The ide is not really up to reconstructing builds.

**Pull request merged**